package-res/resources/web/pentaho/visual/role/mapping.js

```diff
@@ -300,10 +302,10 @@ define([
        *
        * Validity is determined as follows:
        *
-       * 1. if the mapping has no owner visual model it is invalid
+       * 1. if the mapping has no owner visual model, it is invalid
        * 2. if the visual model has a `null` [data]{@link pentaho.visual.base.Model#data},
        *    then every data property in [attributes]{@link pentaho.visual.role.Mapping#attributes} is
-       *    considered undefined.
+       *    considered undefined and invalid.
        * 2. otherwise, if the visual model has a non-`null` [data]{@link pentaho.visual.base.Model#data},
        *    then each data property in [attributes]{@link pentaho.visual.role.Mapping#attributes}:
        *   1. must be defined in `data`
```

Comment:
Suggest eliminating the single bullet and incorporating the contents:

 *   must be defined in `data`